### PR TITLE
L0 provider: do not accept device handle for HOST memory type

### PR DESCRIPTION
**Pull request by igchor**
and do not call zeDeviceGetProperties for HOST provider.

Also, add extra checks for parameters.

### CODE REVIEW

**Review by bratpiorka — APPROVED** [review]

File: test/providers/provider_level_zero.cpp [hunk 1]
```diff
@@ -19,6 +19,89 @@
 using umf_test::test;
 using namespace umf_test;
 
+struct LevelZeroProviderInit
+    : public test,
+      public ::testing::WithParamInterface<umf_usm_memory_type_t> {};
+
+INSTANTIATE_TEST_SUITE_P(, LevelZeroProviderInit,
```
Comment (lukaszstolarczuk) [inline test/providers/provider_level_zero.cpp:26]: redundant, dangling `,` at the beginning...? or is it some magic I don't understand? 😉

File: src/provider/provider_level_zero.c [hunk 2]
```diff
@@ -205,7 +226,9 @@ static umf_result_t ze_memory_provider_alloc(void *provider, size_t size,
     case UMF_MEMORY_TYPE_DEVICE: {
         ze_device_mem_alloc_desc_t dev_desc = {
             .stype = ZE_STRUCTURE_TYPE_HOST_MEM_ALLOC_DESC,
```
Comment (kswiecicki) [inline src/provider/provider_level_zero.c:228]: This should probably also be a `ZE_STRUCTURE_TYPE_DEVICE_MEM_ALLOC_DESC`.

File: src/provider/provider_level_zero.c [hunk 3]
```diff
@@ -220,8 +243,10 @@ static umf_result_t ze_memory_provider_alloc(void *provider, size_t size,
             .pNext = NULL,
             .flags = 0};
         ze_device_mem_alloc_desc_t dev_desc = {
-            .stype = ZE_STRUCTURE_TYPE_DEVICE_MEM_ALLOC_DESC,
-            .pNext = useRelaxedAllocationFlag ? &relaxed_desc : NULL,
+            .stype = ZE_STRUCTURE_TYPE_HOST_MEM_ALLOC_DESC,
```
[the remark this answers is not in the document]
Comment (igchor) [reply src/provider/provider_level_zero.c]: Yes, of course, done.

File: src/provider/provider_level_zero.c (hunk 2 above)
Comment (igchor) [reply src/provider/provider_level_zero.c]: Done.

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**